src/java.base/share/classes/java/net/ServerSocket.java

```diff
@@ -694,7 +694,7 @@ private void implAccept(SocketImpl si) throws IOException {
             Thread thread = Thread.currentThread();
             if (thread.isVirtual() && thread.isInterrupted()) {
                 close();
-                throw new SocketException("Closed by interrupt");
+                throw new SocketException("Closed by interrupt", e);
```

Comment:
The InterruptedIOException is used by the SocketImpl to communicate the interrupt to the ServerSocket. It should not be added a cause here. Same thing in the Socket code.

Reply:
Hi Alan, I removed this one.